### ISSUE TITLE
Feature Selection til classification

### ISSUE THREAD

**Issue by dkalaxdk** [P8-Team/App#24]
- [x] Udvælg værktøj til at arbejde med features i python (Evt tsfresh)
- [x] Dokumenter feature selection prosessen (Til internt brug)

Målet er at uvælge features der hæver hvor akurat den enelige model er uden at overfit den.

**Comment by soenderby** [comment]
*Conclusion*b
[tsfresh](https://github.com/blue-yonder/tsfresh) is a library for automatically selecting and extracting features from time series datasets.
A very nice example of how the library is used can be seen [here.](https://github.com/blue-yonder/tsfresh/blob/main/notebooks/examples/01%20Feature%20Extraction%20and%20Selection.ipynb)
Using this all we need to do is create a pandas [DataFrame](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html) with the format described [here](https://tsfresh.readthedocs.io/en/latest/text/data_formats.html). 
As can be seen in the example, it is also necessary to create a vector containing the class labels for the training data.


One thing to note is that I believe the details of what the library does are beyond the scope of our project. This means that we will likely be unable to explain how/why the specific features are created/selected in any meaningful detail.

The library is also described in this [paper](https://www.sciencedirect.com/science/article/pii/S0925231218304843) which can be cited using the following bibtex.
@article{CHRIST201872,
title = {Time Series FeatuRe Extraction on basis of Scalable Hypothesis tests (tsfresh – A Python package)},
journal = {Neurocomputing},
volume = {307},
pages = {72-77},
year = {2018},
issn = {0925-2312},
doi = {https://doi.org/10.1016/j.neucom.2018.03.067},
url = {https://www.sciencedirect.com/science/article/pii/S0925231218304843},
author = {Maximilian Christ and Nils Braun and Julius Neuffer and Andreas W. Kempa-Liehr},
keywords = {Feature engineering, Time series, Feature extraction, Feature selection, Machine learning},
}

The algorithm used is also described in this [paper](https://arxiv.org/abs/1610.07717) which can be cited using: 
@article{DBLP:journals/corr/ChristKF16,
  author    = {Maximilian Christ and
               Andreas W. Kempa{-}Liehr and
               Michael Feindt},
  title     = {Distributed and parallel time series feature extraction for industrial
               big data applications},
  journal   = {CoRR},
  volume    = {abs/1610.07717},
  year      = {2016},
  url       = {http://arxiv.org/abs/1610.07717},
  eprinttype = {arXiv},
  eprint    = {1610.07717},
  timestamp = {Sat, 23 Jan 2021 01:12:57 +0100},
  biburl    = {https://dblp.org/rec/journals/corr/ChristKF16.bib},
}

The entirety of the documentation can be found (here)[https://docs.google.com/document/d/1BR4CfXoIzIiIqCOupydBDPtKQCnTm0B-cph9vsiJHQk/edit#] (note: it may not be very well structured)

**Comment by soenderby** [comment]
*What is feature selection?*

Basic explanation courtesy of [wikipedia](https://en.wikipedia.org/wiki/Feature_selection):
“In machine learning and statistics, feature selection, also known as variable selection, attribute selection or variable subset selection, is the process of selecting a subset of relevant features (variables, predictors) for use in model construction. “

Also mentions how it differs from feature extraction:
“Feature selection techniques should be distinguished from feature extraction. Feature extraction creates new features from functions of the original features, whereas feature selection returns a subset of the features.”

*How is it done?*
Math and/or magic.
There are numerous algorithms and ways to analyze features to determine if they are relevant.
